go-runtime/schema/analyzers/typealias.go

```diff
@@ -15,13 +14,19 @@ import (
 )
 
 // TypeAliasExtractor extracts type aliases to the module schema.
-var TypeAliasExtractor = &analysis.Analyzer{
-	Name:             "typealias",
-	Doc:              "extracts type aliases to the module schema",
-	Run:              extractTypeAliases,
-	Requires:         []*analysis.Analyzer{inspect.Analyzer},
-	ResultType:       reflect.TypeFor[result](),
-	RunDespiteErrors: true,
+var TypeAliasExtractor = newExtractor("typealias", extractTypeAliases, (*typeAliasExtractorFact)(nil))
+
+type typeAliasExtractorFact struct {
+	decl schema.Decl
+}
+
+func (t *typeAliasExtractorFact) AFact() {}
+func (t *typeAliasExtractorFact) setValue(decl schema.Decl) {
```

Comment:
Should we just call these `Set()` and Get()`?

This is an example of what I meant above BTW - because all these methods are private, it's not clear what their intent is - are they for internal use in a single extractor, or shared between extractors?